dev/TeachingTip/TeachingTip.cpp

```diff
@@ -867,6 +968,17 @@ void TeachingTip::OnActionButtonClicked(const winrt::IInspectable&, const winrt:
 void TeachingTip::OnPopupOpened(const winrt::IInspectable&, const winrt::IInspectable&)
 {
     StartExpandToOpen();
+
+    if (auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*this))
+    {
+        if (auto teachingTipPeer = peer.as<winrt::TeachingTipAutomationPeer>())
+        {
+            winrt::get_self<TeachingTipAutomationPeer>(teachingTipPeer)->RaiseWindowOpenedEvent(
+                ResourceAccessor::GetLocalizedStringResource(SR_TeachingTipNotification) + L" "
+                + winrt::ApplicationModel::Package::Current().DisplayName() + L", "
```

Comment:
strings should not be concatenated like this, this is not friendly for localization. We should have the localized string be the format string. Looks like PersonPicture has implemented this pattern -- take a look at BadgeIconTextOverride and similar strings.

Reply:
I used PersonPicture as a reference but the string format isn't working and the pretty obtuse to debug, any suggestions from here? The resulting string does not include either of the appName or teachingTipAutomationName

Reply:
You need .data() because FormatString takes VA_ARGS and assumes that the thing you're passing in is a PCWSTR.

Reply:
That was it, thank you.